### PR TITLE
MAGECLOUD-3258: Cache Warm-up, Using RegEx to Warm-up Multiple Pages

### PR DESCRIPTION
**Pull request by oshmyheliuk**
https://magento2.atlassian.net/browse/MAGECLOUD-3258

### Related PR

https://github.com/magento/ece-tools/pull/505

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento-cloud-components?pullRequest=2) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento-cloud-components?pullRequest=2) before we can accept your contribution.<br/>

File: Console/Command/ConfigShowEntityUrlsCommand.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Magento\UrlRewrite\Service\V1\Data\UrlRewrite;
+
+/**
+ * Returns list of category or cms-page urls for given stores
+ */
+class ConfigShowEntityUrlsCommand extends Command
+{
+    const INPUT_OPTION_STORE_ID = 'store-id';
```
[the remark this answers is not in the document]
Comment (oshmyheliuk) [reply Console/Command/ConfigShowEntityUrlsCommand.php]: added

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (shiftedreality) [inline .gitignore]: add composer.lock

File: Console/Command/ConfigShowEntityUrlsCommand.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (shiftedreality) [reply Console/Command/ConfigShowEntityUrlsCommand.php]: Magento no more uses this comment annotation

File: Console/Command/ConfigShowEntityUrlsCommand.php [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            'Store ID'
+        );
+        $this->addOption(
+            self::INPUT_OPTION_ENTITY_TYPE,
+            null,
+            InputOption::VALUE_REQUIRED,
+            'Entity type: ' . implode(',', $this->possibleEntities)
+        );
+
+        $this->setName('config:show:urls')
```
Comment (shiftedreality) [inline Console/Command/ConfigShowEntityUrlsCommand.php]: Move to top of configure(), method

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "magento/magento-cloud-components",
+  "description": "Cloud Components Module for Magento 2.x",
+  "type": "magento2-module",
+  "version": "1.0.0",
+  "autoload": {
```
Comment (shiftedreality) [inline composer.json]: Add require Magento modules and `ext-json`

File: Console/Command/ConfigShowStoreUrlCommand.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            'Store ID'
+        );
+        $this->addOption(
+            self::INPUT_OPTION_ALL,
+            'a',
+            InputOption::VALUE_NONE,
+            'Return urls for all stores'
+        );
+
+        $this->setName('config:show:store-url')
```
Comment (shiftedreality) [inline Console/Command/ConfigShowStoreUrlCommand.php]: Also, move to top

File: composer.json (hunk 4 above)
Comment (NadiyaS) [reply composer.json]: not only require. And also suggest

File: .gitignore (hunk 2 above)
Comment (oshmyheliuk) [reply .gitignore]: added

File: Console/Command/ConfigShowEntityUrlsCommand.php (hunk 3 above)
Comment (oshmyheliuk) [reply Console/Command/ConfigShowEntityUrlsCommand.php]: done

File: Console/Command/ConfigShowStoreUrlCommand.php (hunk 5 above)
Comment (oshmyheliuk) [reply Console/Command/ConfigShowStoreUrlCommand.php]: Done

File: composer.json (hunk 4 above)
Comment (oshmyheliuk) [reply composer.json]: added

File: Console/Command/ConfigShowEntityUrlsCommand.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (oshmyheliuk) [reply Console/Command/ConfigShowEntityUrlsCommand.php]: fixed

File: Console/Command/ConfigShowStoreUrlCommand.php [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                'Shows store base url for given id.'
+            );
+
+        $this->addArgument(
+            self::INPUT_ARGUMENT_STORE_ID,
+            InputArgument::OPTIONAL,
+            'Store ID'
+        );
+        $this->addOption(
+            self::INPUT_OPTION_ALL,
```
Comment (NadiyaS) [inline Console/Command/ConfigShowStoreUrlCommand.php]: this option conflicts with argument.
What if I use both? this behavior should be described. 
I propose show URLs for all stores id special ID was not set

File: Console/Command/ConfigShowStoreUrlCommand.php [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $this->setName('config:show:store-url')
+            ->setDescription(
+                'Shows store base url for given id.'
```
Comment (NadiyaS) [inline Console/Command/ConfigShowStoreUrlCommand.php]: change description, that for base URLs will be shown if id was not passed

Comment (oshmyheliuk) [reply Console/Command/ConfigShowStoreUrlCommand.php]: Done

**Comment by NadiyaS** [comment]
As it is not convenient to add comments to the Zephyr test, I will add my scenarios here which should be included to test:
 - if we have 5 websites with 2 store views for each of them, will command command `config:show:store-url` return all 10 URLs?
 - we have one website with defined URL. this website has several store views where URL is the same as for website. What URL will be displayed for every store (it should have ___store in the path)

**Comment by oshmyheliuk** [comment]
- if we have 5 websites with 2 store views for each of them, will command command config:show:store-url return all 10 URLs?

Yes, the command will display URL for each store.

- we have one website with defined URL. this website has several store views where URL is the same as for website. What URL will be displayed for every store (it should have ___store in the path)

In this case, URLs will be the same for each store, maybe it requires some additional configuration to add `___store` in the path?

**Comment by NadiyaS** [comment]
@oshmyheliuk I guess we need to add option (or something) to retrieve URL which lead to real store view page and not to default store view

**Pull request merged**